### PR TITLE
Hookup form to post endpoint

### PR DESCRIPTION
**Pull request by nate-kirschner**
### ℹ️ Issue

Closes #<issue number>

### 📝 Description

I integrated the /form endpoint into the forms page and synced the input validation on the frontend and backend

I also refactored some of the frontend components
- New Form component which combines the grant and medical form since they will be submitted at the same time and shouldn't have two different submit buttons (for now)
- Made a FormField component which abstracts some of the FormControl logic and repeated code
- Made a FormInput component which takes a inputVariant prop and determines which chakra ui input type to display

### ✔️ Verification

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves. 

Provide screenshots of any new components, styling changes, or pages. 

### 🏕️ (Optional) Future Work / Notes

Did you notice anything ugly during the course of this ticket? Any bugs, design challenges, or unexpected behavior? Write it down so we can clean it up in a future ticket!

### CODE REVIEW

File: apps/backend/src/handlers/getAllForms.ts [hunk 1]
```diff
@@ -6,27 +6,44 @@ import { fetchDocuments } from '../db/utils.js';
...
+    console.info(
+      `response from: ${event.path} statusCode: ${response.statusCode}`
+    );
 
-  return response;
-};
\ No newline at end of file
+    return response;
+  } catch (error) {
+    console.log('error accessing database');
```
Comment (megandouli) [inline apps/backend/src/handlers/getAllForms.ts]: can you make this `console.error(error)`? So we get the whole error log in Cloudwatch if something goes wrong

**Review by megandouli — DISMISSED** [review]

File: apps/backend/src/handlers/getAllForms.ts (hunk 1 above)
Comment (megandouli) [reply apps/backend/src/handlers/getAllForms.ts]: also were you able to test this full-stack?

**Review by megandouli — COMMENTED** [review]
Some questions / concerns (might need input from @bellehlim or @jackielincroft):

How will the child’s parent / legal guardian complete part of the form and the medical professional complete the other part if the two forms are on the same page and there's no unique URL to identify a partially completed form?

Doctor’s Name and Doctor’s Title seem to be required fields for the backend (form submission error when I left them blank) but they’re not marked as required on the frontend

File: apps/backend/src/handlers/getAllForms.ts (hunk 1 above)
Comment (nate-kirschner) [reply apps/backend/src/handlers/getAllForms.ts]: Yep deployed it on my aws account and tested it

**Comment by nate-kirschner** [comment]
@megandouli 

> How will the child’s parent / legal guardian complete part of the form and the medical professional complete the other part if the two forms are on the same page and there's no unique URL to identify a partially completed form?

Yeah I wasn't sure about this as well

> Doctor’s Name and Doctor’s Title seem to be required fields for the backend (form submission error when I left them blank) but they’re not marked as required on the frontend

Hmm I'm pretty sure they're marked as optional on the backend. Are you sure you didn't have another required field that was blank?

**Comment by megandouli** [comment]
> @megandouli
> 
> > How will the child’s parent / legal guardian complete part of the form and the medical professional complete the other part if the two forms are on the same page and there's no unique URL to identify a partially completed form?
> 
> Yeah I wasn't sure about this as well
> 
> > Doctor’s Name and Doctor’s Title seem to be required fields for the backend (form submission error when I left them blank) but they’re not marked as required on the frontend
> 
> Hmm I'm pretty sure they're marked as optional on the backend. Are you sure you didn't have another required field that was blank?

Yeah I just tried it, leaving those two fields empty results in a form submission error :(

File: apps/backend/src/handlers/getAllForms.ts [hunk 2]
```diff
@@ -6,27 +6,44 @@ import { fetchDocuments } from '../db/utils.js';
...
+    console.info(
+      `response from: ${event.path} statusCode: ${response.statusCode}`
+    );
 
-  return response;
-};
\ No newline at end of file
+    return response;
+  } catch (error) {
+    console.error('error accessing database');
```
Comment (megandouli) [inline apps/backend/src/handlers/getAllForms.ts]: oh can you also include `error` in the log?

**Comment by jackielincroft** [comment]
> How will the child’s parent / legal guardian complete part of the form and the medical professional complete the other part if the two forms are on the same page and there's no unique URL to identify a partially completed form?

Yeah this is a good question -- I know often times a hospital social worker will fill out the first form on behalf of the family and that person might be able to fill in both forms at once, but it does seem like there might be a case where those should be filled out separately.  We'll check with Lori!!

**Review by megandouli — DISMISSED** [review]
awesome ty!

**Review by megandouli — APPROVED** [review]

**Pull request merged**